codec/enforceproto/extraneous_fields.go

```diff
@@ -0,0 +1,200 @@
+package enforceproto
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	gogoproto "github.com/gogo/protobuf/proto"
+	"google.golang.org/protobuf/encoding/protowire"
+	"google.golang.org/protobuf/proto"
+)
+
+const bit11Reserved = 1 << 10
```

Comment:
Let's use the language non-critical throughout this code instead of reserved to be consistent with the language in ADR 020. Also reserved has a different meaning already in protobuf and we don't want to create confusion. I wouldn't consider these non-critical fields "reserved".

```suggestion
const bit11NonCritical = 1 << 10
```

Reply:
Ack, done and thanks!